### ISSUE TITLE
No route to host exception while running tests

### ISSUE THREAD

**Issue by aloiscochard** [scalatest/scalatest#248]
```
> last service/test:testOnly
[error] Uncaught exception when running tests: java.net.NoRouteToHostException: No route to host
sbt.ForkMain$ForkError: No route to host
        at java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:339)
        at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
        at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
        at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.net.Socket.connect(Socket.java:579)
        at java.net.Socket.connect(Socket.java:528)
        at java.net.Socket.<init>(Socket.java:425)
        at java.net.Socket.<init>(Socket.java:208)
        at org.scalatest.tools.SocketReporter.<init>(SocketReporter.scala:26)
        at org.scalatest.tools.ReporterFactory.createSocketReporter(ReporterFactory.scala:221)
        at org.scalatest.tools.ReporterFactory.getReporterFromConfiguration(ReporterFactory.scala:236)
        at org.scalatest.tools.ReporterFactory$$anonfun$createReportersFromConfigurations$1.apply(ReporterFactory.scala:242)
        at org.scalatest.tools.ReporterFactory$$anonfun$createReportersFromConfigurations$1.apply(ReporterFactory.scala:241)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
        at scala.collection.Iterator$class.foreach(Iterator.scala:727)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1157)
        at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
        at org.scalatest.tools.ReporterConfigurations.foreach(ReporterConfiguration.scala:43)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
        at org.scalatest.tools.ReporterConfigurations.map(ReporterConfiguration.scala:43)
        at org.scalatest.tools.ReporterFactory.createReportersFromConfigurations(ReporterFactory.scala:241)
        at org.scalatest.tools.ReporterFactory.getDispatchReporter(ReporterFactory.scala:245)
        at org.scalatest.tools.Framework$ScalaTestRunner.<init>(Framework.scala:808)
        at org.scalatest.tools.Framework.runner(Framework.scala:1106)
        at sbt.ForkMain$Run.runTests(ForkMain.java:186)
        at sbt.ForkMain$Run.run(ForkMain.java:257)
        at sbt.ForkMain.main(ForkMain.java:99)
```

**Comment by pcleary00** [comment]
I am hitting this now as well.  I added the "junit-interface" as I need to run junit tests as well.  Once I started doing this, I started getting this exception.

Running:

- Scalatest 2.2.3
- Scala 2.10.4
- Mockito 1.9.5
- Junit-interface 0.11
- sbt 0.13.5
- Mac Yosemite

Any update on when this will be in a release?

**Comment by bvenners** [comment]
OK. I want to get a 2.3 release out before nescala, which starts Friday, Jan 30. So it will be this week. Sorry for the delay. Was a busy several months.

**Comment by myyk** [comment]
@pcleary00 You probably can use the 2.2.3-SNAP2 release as a workaround until @bvenners gets the release out which sounds like later this week. I've been using it for most of the projects I've been touching since I need to be on VPN a lot of times while running tests and have no problems with it for the last week.

**Comment by bvenners** [comment]
Hi All,

I verified last night that indeed we somehow didn't get the fix into 2.2.3. Sorry about that. That's what 2.2.3 was supposed to be for. We put it into the branch that we'll be building 2.2.4 from as soon as we get a few more commits in there.

https://github.com/scalatest/scalatest/commit/504583323e4eba2e37525cde41e52ccc146899fd

Bill

**Comment by bvenners** [comment]
This was fixed in 2.2.4.

**Comment by SyedJameer** [comment]
Exception in thread "main" java.net.NoRouteToHostException: No route to host
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:339)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:579)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:123)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:133)
	at org.apache.http.impl.conn.AbstractPoolEntry.open(AbstractPoolEntry.java:149)
	at org.apache.http.impl.conn.AbstractPooledConnAdapter.open(AbstractPooledConnAdapter.java:108)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:415)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:641)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:576)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:554)
	at example.producer.Spout6.main(Spout6.java:79)
